src/main/kotlin/ecommerce/service/ProductService.kt

```diff
@@ -4,22 +4,39 @@ import ecommerce.dto.ProductForm
 import ecommerce.exception.InternalServerErrorException
 import ecommerce.exception.NotFoundException
 import ecommerce.exception.ProductNameAlreadyExistsException
+import ecommerce.model.Option
 import ecommerce.model.Product
+import ecommerce.repository.OptionRepository
 import ecommerce.repository.ProductRepository
+import org.springframework.data.domain.Page
+import org.springframework.data.domain.PageRequest
+import org.springframework.data.domain.Sort
 import org.springframework.data.repository.findByIdOrNull
 import org.springframework.stereotype.Service
 
 @Service
-class ProductService(private val productRepository: ProductRepository) {
+class ProductService(
+    private val productRepository: ProductRepository,
+    private val optionRepository: OptionRepository,
+) {
     fun insert(form: ProductForm): Product {
-        checkProductNameExists(form.name)
-        val product = ProductForm.toProduct(form)
+        nameExists(form.name)
+        val option = Option(name = "none", quantity = 1)
+        val product = ProductForm.toProduct(form, listOf(option))
+        option.product = product
         val savedProduct = productRepository.save(product)
+        optionRepository.save(option)
         return productRepository.findByIdOrNull(savedProduct.id)
             ?: throw InternalServerErrorException("ProductService.insert() - Product with ID ${savedProduct.id} not found")
     }
```

Comment:
If `savedProduct` has been returned, that means it’s already created, so I don’t think we need to look it up again. What do you think? 🤔